packages/labs/context/src/lib/decorators/provide.ts

```diff
@@ -84,16 +84,33 @@ export function provide<ValueType>({
         protoOrTarget,
         nameOrContext
       );
-      const oldSetter = descriptor?.set;
-      const newDescriptor = {
-        ...descriptor,
-        set: function (this: ReactiveElement, value: ValueType) {
-          controllerMap.get(this)?.setValue(value);
-          if (oldSetter) {
-            oldSetter.call(this, value);
-          }
-        },
-      };
+      let newDescriptor;
+      if (descriptor === undefined) {
+        const symbol = Symbol();
+        newDescriptor = {
+          get: function (this: ReactiveElement & {[symbol]: ValueType}) {
+            return this[symbol];
+          },
+          set: function (
+            this: ReactiveElement & {[symbol]: ValueType},
+            value: ValueType
+          ) {
+            controllerMap.get(this)?.setValue(value);
```

Comment:
This element will always be in the map by this point, so I don't think we need to optional chaining.

Reply:
Switched to not-null assertion